lib/glimesh_web/live/gct_live/components/button_array.ex

```diff
@@ -0,0 +1,132 @@
+defmodule GlimeshWeb.GctLive.Components.ButtonArray do
+  use GlimeshWeb, :live_view
+
+  alias Glimesh.Accounts
+  alias Glimesh.CommunityTeam
+
+  @impl true
+  def render(assigns) do
+    ~L"""
+    <%= live_redirect gettext("Edit Profile"), class: (if @can_edit_profile, do: "btn btn-primary", else: "btn btn-primary disabled"), to: Routes.gct_path(@socket, :edit_user_profile, @user.username) %>
+    <%= live_redirect gettext("Edit User"), class: (if @can_edit_user, do: "btn btn-primary", else: "btn btn-primary disabled"), to: Routes.gct_path(@socket, :edit_user, @user.username) %>
+    <%= unless @user.is_banned do %>
+      <button class="btn btn-danger" phx-click="show_ban_modal" <%= unless @can_ban, do: "disabled" %> ><%= gettext("Ban User") %></button>
+    <% else %>
+      <button class="btn btn-danger" phx-click="unban_user" <%= unless @can_ban, do: "disabled" %> ><%= gettext("Unban User") %></button>
+    <% end %>
+
+    <%= if @show_ban do %>
+      <div id="ban-modal" class="live-modal"
+        phx-capture-click="hide_ban_modal"
+        phx-window-keydown="hide_ban_modal"
+        phx-key="escape"
+        phx-target="#paymentModal2"
+        phx-page-loading>
+        <div class="modal-dialog" role="document">
+          <div class="modal-content">
+            <div class="modal-header">
+              <h5 class="modal-title"><%= gettext("Ban User") %></h5>
+              <button type="button" class="close" phx-click="hide_ban_modal" aria-label="Close">
+                <span aria-hidden="true">&times;</span>
+              </button>
+            </div>
+
+            <div class="modal-body">
+              <%= form_for :user, "#", [phx_submit: :ban] %>
+              <div class="form-group ">
+                <label for="banReason"><%= gettext("Ban Reason") %></label>
+                <textarea rows="5" class="form-control" name="ban_reason" id="banReason" placeholder="A descriptive reason to why this user is being banned. This is required, if you don't provide one then the user will not be banned."></textarea>
+              </div>
+
+              <button class="btn btn-danger btn-block mt-4"><%= gettext("Ban User") %></button>
+            </div>
+          </div>
+        </div>
+      </div>
+    <% end %>
+    """
+  end
+
+  @impl true
+  def mount(_params, %{"admin" => admin, "user" => user}, socket) do
+    can_ban =
+      case Bodyguard.permit(Glimesh.CommunityTeam, :can_ban, admin, user),
+        do:
+          (
+            :ok -> true
+            {:error, :unauthorized} -> false
+          )
```

Comment:
Can be shortened to: 
```suggestion
    can_ban = Bodyguard.permit?(Glimesh.CommunityTeam, :can_ban, admin, user)
```

same with below can_'s